### PR TITLE
134 fix psl args

### PR DESCRIPTION
**Pull request by Matthew-Grayson**
#  #
 
## 🗣 Description ##

`--psl-filename=FILENAME` and `--psl-read-only` now work as advertised.

Remove monkey patches in domain.py. Remove PublicSuffixListFilename / PublicSuffixListReadOnly  initialization in \_\_main\_\_.py; set default values in docopt usage pattern to compensate. Pass filename and readonly as a json file between cli.py and domain.py.

Additionally, removed several constants/variables from cli.py that are no longer needed after updating usage pattern. Trustymail now prints the docopt args dictionary. This is useful for confirming the default values are being used (feel free to remove this after testing).

## 💭 Motivation and context ##

This closes issue #134 

## 🧪 Testing ##

1. Create fresh python environment
2. Run `pip install git+https://github.com/Matthew-Grayson/trustymail@134-fix-psl-args`
3. Run `trustymail cisa.gov`
4. Confirm that default values are correct in statement printed to cli
5. Run `trustymail --psl-filename=testfilename.dat cisa.gov`
6. Confirm creation of testfilename.dat
7. If needed, cd to directory of testfilename.dat
8. Change mtime of psl by running `python3 -c "from os import utime; os.utime('testfilename.dat', (1330712280, 1330712292))"`
9. Confirm that last modified time of testfilename.dat is changed
10. Run `trustymail --psl-filename=testfilename.dat --psl-read-only cisa.gov`
11. Confirm testfilname.dat is not overwritten
12. Run `trustymail --psl-filename=testfilename.dat cisa.gov`
13. Confirm that testfilename.dat is overwritten

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: src/trustymail/cli.py [hunk 1]
```diff
@@ -61,20 +62,16 @@
...
-
 
 def main():
     """Perform a trustymail scan using the provided options."""
     args = docopt.docopt(__doc__, version=__version__)
+    print('args as interpreted by trustymail:', args)
+
+    # Write the arguments to a file for use by the trustymail library
+    with open('env.json', 'w') as env:
+        json.dump(args, env)
```
Comment (mcdonnnj) [inline src/trustymail/cli.py:73]: This is worse than a monkey patch and a complete non-starter to me as it breaks the API of this package.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: src/trustymail/cli.py (hunk 1 above)
Comment (Matthew-Grayson) [reply src/trustymail/cli.py:73]: Noted; I'll make an adjustment.

Comment (Matthew-Grayson) [reply src/trustymail/cli.py:73]: @mcdonnnj Thanks for the feedback. What are your thoughts on adding arguments (readonly, filename) to domain.get_psl  and moving the call to get_psl to cli.py?

Comment (mcdonnnj) [reply src/trustymail/cli.py:73]: I need to look at the least destructive way to fix this as I believe the monkey patch may only be an issue for the CLI. This package is used in other projects so I lean very conservatively against breaking changes that are not strictly necessary.

**Pull request closed without merging**